service/src/main/java/com/epam/esm/gifts/impl/UserServiceImpl.java

```diff
@@ -3,135 +3,140 @@
 import com.epam.esm.gifts.UserService;
 import com.epam.esm.gifts.converter.OrderConverter;
 import com.epam.esm.gifts.converter.UserConverter;
-import com.epam.esm.gifts.dao.impl.UserDaoImpl;
+import com.epam.esm.gifts.dao.OrderRepository;
+import com.epam.esm.gifts.dao.UserRepository;
 import com.epam.esm.gifts.dto.CustomPage;
-import com.epam.esm.gifts.dto.CustomPageable;
 import com.epam.esm.gifts.dto.ResponseOrderDto;
 import com.epam.esm.gifts.dto.UserDto;
 import com.epam.esm.gifts.exception.SystemException;
+import com.epam.esm.gifts.model.Order;
 import com.epam.esm.gifts.model.User;
 import com.epam.esm.gifts.validator.EntityValidator;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import java.util.List;
 import java.util.Optional;
 
+import org.springframework.security.crypto.password.PasswordEncoder;
+import static com.epam.esm.gifts.dto.UserDto.Role.USER;
 import static com.epam.esm.gifts.exception.ExceptionCode.*;
 
 @Service
 public class UserServiceImpl implements UserService {
 
 
-    private UserDaoImpl userDao;
+    private UserRepository userRepository;
     private EntityValidator validation;
     private UserConverter userConverter;
     private OrderConverter orderConverter;
+    private OrderRepository orderRepository;
+    private PasswordEncoder passwordEncoder;
 
     @Autowired
-    public UserServiceImpl(UserDaoImpl userDao, EntityValidator validation,
-                           UserConverter userConverter, OrderConverter orderConverter) {
-        this.userDao = userDao;
+    public UserServiceImpl(UserRepository userRepository, EntityValidator validation,
+                           UserConverter userConverter, OrderConverter orderConverter,
+                           PasswordEncoder passwordEncoder, OrderRepository orderRepository) {
+        this.userRepository = userRepository;
         this.validation = validation;
         this.userConverter = userConverter;
         this.orderConverter = orderConverter;
+        this.passwordEncoder = passwordEncoder;
+        this.orderRepository = orderRepository;
     }
 
     @Override
     @Transactional
```

Comment:
Can Transactional be added to the class instead of methods?

Reply:
possible, but @Transactional in a class applies to each service method and have some differences